src/type/s2k/generic.ts

```diff
@@ -123,24 +113,19 @@ class GenericS2K {
    * Serializes s2k information
    * @returns {Uint8Array} Binary representation of s2k.
    */
-  write() {
-    if (this.type === 'gnu-dummy') {
-      return new Uint8Array([101, 0, ...util.stringToUint8Array('GNU'), 1]);
-    }
+  write(): Uint8Array {
     const arr = [new Uint8Array([enums.write(enums.s2k, this.type), this.algorithm])];
 
     switch (this.type) {
       case 'simple':
         break;
       case 'salted':
-        arr.push(this.salt);
+        this.salt && arr.push(this.salt);
```

Comment:
Better to throw here (same for the 'iterated' case that follows), or I think it gives the wrong impression that the salt is optional 🙂 

```suggestion
        if (!salt) { throw Error('Salt was not set') }
        arr.push(this.salt);
```